Add onboarding preparation docs (#2341)
sourcegraph/about: handbook/people-ops/onboarding/before_starting.md

Recent and planned improvements are noted in _italics_ in the list below.

## People Ops onboarding

### Immediately after offer signing

- Each new teammate is added to [teammate tracking spreadsheet](https://docs.google.com/spreadsheets/d/1qt3rGNRVaSyi_N2uN3DaXvlg8LjfKLJb8PFfrQsmR-I/edit#gid=948830545) (name, title, start date, location)

### One week prior to first day

- Each new teammate receives a welcome email a week before their start date providing information and asking questions about benefits, email address selection, equipment purchases, and other onboarding logistics (we provide two templates, for US and non-US teammates)
  - [US template](https://docs.google.com/document/d/1a3uhkt0bvmMuTmFBvroEXSdQGIPwCz8BLgk_YiSMZUQ/edit)
  - [Non-US template](https://docs.google.com/document/d/1OxrGknE6NKoeMY7qMDe5th8ZL3juxFOKfH-eywWuwko/edit)

>   - _As of 2021-01-18, all hiring managers are CCed on these emails_

Reviewer: 🎉

Reviewer 2: I was making a couple of edits to our buddy page but those are done now and I'll add in that section here too!

  - [US template - Google Tables](https://tables.area120.google.com/u/0/workspace/ahgZbSufPWldLJBbYNYk9F?utm_source=share&utm_medium=referral&utm_campaign=workspacelink)
  - [Non-US - Asana](https://app.asana.com/-/login?u=%2F0%2F1199579360338646%2F1199579360338646)
  - _One of our first onboarding issues to tackle in the coming weeks is getting onboarding checklists into a more cohesive state that bundles general onboarding + team/role specific onboarding, and does it in one place vs. a combination of tools + handbook_
    - _Google Tables’ layout and model is more ideal, but it is in beta and only available in the US right now, so we will continue to use Asana for non-US teammates until this issue is resolved_
- Each new teammate is added to a short list of general onboarding accounts
  - Gmail + any specific mailing groups
    - _As of 2021-01-15, all hiring managers will be notified of the email addresses going live upon creation_
  - Slack + any specific channels and/or teams
  - GitHub (if they already have an account)
  - 1Password
  - Zoom
  - Lever
  - Lattice

> - New hires are assigned an onboarding buddy by their manager (People Ops will reach out to managers via Slack to confirm onboarding buddy)

Author: ```suggestion
- Each new teammate is assigned an onboarding buddy by their manager (People Ops will reach out to managers via Slack to confirm the onboarding buddy)
```
